### PR TITLE
Product name unification - update headers

### PR DESCRIPTION
**Pull request by mataspetrikas**
after renaming the MQTTFramework.framework to MQTTClient.framework 
update the headers
remove the MQTTFramework.h file
add missing imports in MQTTClient.h

### CODE REVIEW

File: MQTTClient/MQTTFramework/MQTTFramework.h [hunk 1]
```diff
@@ -1,30 +0,0 @@
...
-#import "MQTTSessionSynchron.h"
-#import "MQTTMessage.h"
-#import "MQTTTransport.h"
-#import "MQTTCFSocketTransport.h"
-#import "MQTTCoreDataPersistence.h"
-#import "MQTTSSLSecurityPolicyTransport.h"
-#import "MQTTSessionManager.h"
-
-//! Project version number for MQTTFramework.
-FOUNDATION_EXPORT double MQTTFrameworkVersionNumber;
```
Comment (jcavar) [inline MQTTClient/MQTTFramework/MQTTFramework.h]: Do we need these exports in `MQTTClient.h` now?

Comment (mataspetrikas) [reply MQTTClient/MQTTFramework/MQTTFramework.h]: @jcavar a good question. I'm not entirely sure which compnents consumes these. For the sake of concistency I could add them there

File: MQTTClient/MQTTClient/MQTTClient.h [hunk 2]
```diff
@@ -24,3 +25,11 @@
 #import <MQTTClient/MQTTCFSocketTransport.h>
 #import <MQTTClient/MQTTCoreDataPersistence.h>
 #import <MQTTClient/MQTTSSLSecurityPolicyTransport.h>
+#import <MQTTClient/MQTTSessionManager.h>
+
+//! Project version number for MQTTClient.
+FOUNDATION_EXPORT double MQTTFrameworkVersionNumber;
```
Comment (jcavar) [inline MQTTClient/MQTTClient/MQTTClient.h]: But it should be`MQTTClientVersionNumber` now, right?

**Comment by mataspetrikas** [comment]
@jcavar I updated the naming in version exports plus on targets.

**Pull request merged**